src/network/networkd-setlink.c

```diff
@@ -1097,6 +1101,9 @@ static bool link_is_ready_to_bring_up_or_down(Link *link, bool up) {
         if (!link->activated)
                 return false;
 
+        if (up && link_rfkilled(link) > 0)
```

Comment:
I'm not sure if that's a good idea. IMHO `link_rfkilled()` is somewhat "advisory", if not "cosmetic". Changing the "do that or not" functions (especially when they are not even wifi-specific) for that seems going too far, if not dangerous.

Reply:
Sure, but we should still log about it if something goes wrong, we can still ignore the error higher up the stack

Reply:
The wifi interface may be handled by the old wext driver, instead of ieee80211. In that case, I do not know what is returned by the function. Hence, as @tomty89 commented, the function should be just a kind of advisory.

Reply:
Can we add more debug logging to link_rfkilled itself then, in the error paths?

Reply:
In the revised version, it logged in many failure path.